randomwallpaper@iflow.space/adapter/reddit.js

```diff
@@ -45,6 +45,15 @@ var RedditAdapter = class extends BaseAdapter.BaseAdapter {
 				const submissions = response_body.data.children.filter(child => {
 					if (child.data.post_hint !== 'image') return false;
 					if (require_sfw) return child.data.over_18 === false;
+
+					let minWidth = this._settings.get('min-width', 'int');
+					let minHeight = this._settings.get('min-height', 'int');
+					if (child.data.preview.images[0].source.width < minWidth) return false;
+					if (child.data.preview.images[0].source.height < minHeight) return false;
+
+					let imageRatio1 = this._settings.get('image-ratio1', 'int');
+					let imageRatio2 = this._settings.get('image-ratio2', 'int');
+					if (child.data.preview.images[0].source.width / imageRatio1 * imageRatio2 < child.data.preview.images[0].source.height) return false;
```

Comment:
Hm, I think I need an explanation for these ratio restrictions. I would have expected that you just want something like `if(width/x != height/y) { return false; }` to only allow a specific ratio such as 16:9 (x=16, y=9).

Due to the "smaller than" (`<`), the code would behave a bit unintuitive. If I enter 16:9 it will also allow any image that is wider than 16:9 (e.g., 17:9). Is that desired for some reason?

Also, I think we need to describe the ratio values in the UI a bit for usability (at least display the formula used there)

Reply:
The idea was that allowing a wider range for almost fitting pictures will allow for a more varied pool of pictures. Forcing the exact ratio is quite limiting for Reddit as e.g. for `r/earthporn` those ratios are almost never met.

For more extreme panoramas you'll lose quite substantially on the sides but as the minimum resolutions are met they won't look bad from a quality perspective. E.g. https://old.reddit.com/r/EarthPorn/comments/126glr9/wyomings_wind_river_canyon_is_pretty_spectacular/ will still look good. (bad example because the image quality itself is bad despite a high resolution)

But I can see this failing for portrait displays. Probably needs a second path for those if you want to keep such logic at all.
``` ts
// landscape
if (imageRatio1 > imageRatio2 && child.data.preview.images[0].source.width / imageRatio1 * imageRatio2 < child.data.preview.images[0].source.height)
    return false;
// portrait
else if (imageRatio1 < imageRatio2 && child.data.preview.images[0].source.height / imageRatio2 * imageRatio1 < child.data.preview.images[0].source.width)
    return false;
```

Reply:
Ok, increasing the image pool size makes total sense to me. So I'm fine with that!

Maybe we want to provide a more flexible interface in the future, something like "enter aspect ratio" + "require exact" or "allow wider" or "allow taller" (or something similar). :)